File: фибоначчи.java
public class FibonacciHeap {
    private static class Node {
        int key;
        Node child, left, right;
        int degree;
    }
    
    private Node minNode;
    private int size;
    
    public void insert(int key) { /* ... */ }
    public int extractMin() { /* ... */ }
}

File: фибоначчи.cpp
// Пример структуры узла
struct FibNode {
    int key;
    FibNode* child;
    FibNode* left;
    FibNode* right;
    int degree;
};

class FibonacciHeap {
    FibNode* minNode;
    int count;
public:
    void insert(int key) { /* ... */ }
    int extractMin() { /* ... */ }
};
